### PR TITLE
bugfix/klaviyo-too-many-partitions

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
## PR Overview
**This PR will address the following Issue/Feature:** `v0.7.0`

**This PR will result in the following new package version:** No linked issue as this is a result of an upstream change in the Klaviyo package. See [dbt_klaviyo PR 41](https://github.com/fivetran/dbt_klaviyo/pull/41) for more details.

This is a breaking change due to the upstream changes which require a full refresh for bigquery users.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## Upstream Klaviyo Breaking Changes (Full refresh required after upgrading)
> - Upstream incremental models within the dbt_klaviyo package running on BigQuery have had the `partition_by` logic removed from incremental models running on BigQuery. This change affects only BigQuery warehouses and resolves the `too many partitions` error that some users encountered. The partitioning was also deemed unnecessary for the mentioned models and their downstream references, offering no performance benefit. By removing it, we eliminate both the error risk and an unneeded configuration. Refer to the [v0.8.0 dbt_klaviyo release notes](https://github.com/fivetran/dbt_klaviyo/releases/tag/v0.7.2) for more details. This change applies to the following models:
>   - `int_klaviyo__event_attribution`
>   - `klaviyo__events`
> 
> ## Under the Hood
> - Added consistency and integrity validation tests for the following models:
>   - `shopify_holistic_reporting__customer_enhanced`
>   - `shopify_holistic_reporting__daily_customer_metrics`
>   - `shopify_holistic_reporting__orders_attribution`
> - Cleaned up unnecessary variable configuration within the `integration_tests/dbt_project.yml` file.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [X] dbt run –full-refresh && dbt test
- [X] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [X] The appropriate issue has been linked, tagged, and properly assigned
- [X] All necessary documentation and version upgrades have been applied
    
- [X] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

See below for the results of the validation tests (full refresh and incremental runs)

image

image

### If you had to summarize this PR in an emoji, which would it be?
 
🐟

### CODE REVIEW

File: packages.yml [hunk 1]
```diff
@@ -2,5 +2,8 @@ packages:
...
   version: [">=0.11.0", "<0.14.0"]
 
-- package: fivetran/klaviyo
-  version: [">=0.7.0", "<0.8.0"]
\ No newline at end of file
+# - package: fivetran/klaviyo
+#   version: [">=0.7.0", "<0.8.0"]
+- git: https://github.com/fivetran/dbt_klaviyo.git
+  revision: bugfix/too-many-partitions
+  warn-unpinned: false
```
Comment (fivetran-joemarkiewicz) [inline packages.yml:9]: To update before merge
```suggestion
 - package: fivetran/klaviyo
   version: [">=0.8.0", "<0.9.0"]
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,19 @@
+# dbt_shopify_holistic_reporting v0.7.0
+
+[PR #26](https://github.com/fivetran/dbt_shopify_holistic_reporting/pull/26) includes the following changes:
+
+## Upstream Klaviyo Breaking Changes (Full refresh required after upgrading)
+- Upstream incremental models within the dbt_klaviyo package running on BigQuery have had the `partition_by` logic removed from incremental models running on BigQuery. This change affects only BigQuery warehouses and resolves the `too many partitions` error that some users encountered. The partitioning was also deemed unnecessary for the mentioned models and their downstream references, offering no performance benefit. By removing it, we eliminate both the error risk and an unneeded configuration. Refer to the [v0.8.0 dbt_klaviyo release notes](https://github.com/fivetran/dbt_klaviyo/releases/tag/v0.7.2) for more details. This change applies to the following models:
```
Comment (fivetran-catfritz) [inline CHANGELOG.md:6]:
```suggestion
- Upstream incremental models within the dbt_klaviyo package running on BigQuery have had the `partition_by` logic removed from incremental models running on BigQuery. This change affects only BigQuery warehouses and resolves the `too many partitions` error that some users encountered. The partitioning was also deemed unnecessary for the mentioned models and their downstream references, offering no performance benefit. By removing it, we eliminate both the error risk and an unneeded configuration. Refer to the [v0.8.0 dbt_klaviyo release notes](https://github.com/fivetran/dbt_klaviyo/releases/tag/v0.8.0) for more details. This change applies to the following models:
```

**Review by fivetran-catfritz — APPROVED** [review]
One small change but the rest lgtm!

File: CHANGELOG.md (hunk 2 above)
Comment (fivetran-joemarkiewicz) [reply CHANGELOG.md:6]: Thanks for catching this!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+
+[PR #26](https://github.com/fivetran/dbt_shopify_holistic_reporting/pull/26) includes the following changes:
+
+## Upstream Klaviyo Breaking Changes (Full refresh required after upgrading)
+- Upstream incremental models within the dbt_klaviyo package running on BigQuery have had the `partition_by` logic removed from incremental models running on BigQuery. This change affects only BigQuery warehouses and resolves the `too many partitions` error that some users encountered. The partitioning was also deemed unnecessary for the mentioned models and their downstream references, offering no performance benefit. By removing it, we eliminate both the error risk and an unneeded configuration. Refer to the [v0.8.0 dbt_klaviyo release notes](https://github.com/fivetran/dbt_klaviyo/releases/tag/v0.8.0) for more details. This change applies to the following models:
+  - `int_klaviyo__event_attribution`
+  - `klaviyo__events`
+
+## Under the Hood
+- Added consistency and integrity validation tests for the following models:
```
Comment (fivetran-avinash) [inline CHANGELOG.md:11]:
```suggestion
- Added consistency validation tests for the following models:
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,19 @@
+# dbt_shopify_holistic_reporting v0.7.0
+
+[PR #26](https://github.com/fivetran/dbt_shopify_holistic_reporting/pull/26) includes the following changes:
+
+## Upstream Klaviyo Breaking Changes (Full refresh required after upgrading)
+- Upstream incremental models within the dbt_klaviyo package running on BigQuery have had the `partition_by` logic removed from incremental models running on BigQuery. This change affects only BigQuery warehouses and resolves the `too many partitions` error that some users encountered. The partitioning was also deemed unnecessary for the mentioned models and their downstream references, offering no performance benefit. By removing it, we eliminate both the error risk and an unneeded configuration. Refer to the [v0.8.0 dbt_klaviyo release notes](https://github.com/fivetran/dbt_klaviyo/releases/tag/v0.8.0) for more details. This change applies to the following models:
```
Comment (fivetran-avinash) [inline CHANGELOG.md:6]:
```suggestion
- Upstream incremental models within the dbt_klaviyo package running on BigQuery have had the `partition_by` logic removed. This change affects only BigQuery warehouses and resolves the `too many partitions` error that some users encountered. The partitioning was also deemed unnecessary for the aforementioned models and their downstream references, offering no performance benefit. By removing it, we eliminate both the error risk and an unneeded configuration. Refer to the [v0.8.0 dbt_klaviyo release notes](https://github.com/fivetran/dbt_klaviyo/releases/tag/v0.8.0) for more details. This change applies to the following models:
```

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-joemarkiewicz A few small CHANGELOG revisions.

Before approving, I also noticed that `shopify_holistic_reporting__orders_attribution` ([link](https://github.com/fivetran/dbt_shopify_holistic_reporting/blob/main/models/shopify_holistic_reporting__orders_attribution.sql)) had similar partition by logic that was removed in `dbt_klaviyo`. It seems `created_timestamp` is involved in a join in that model, so it might hold more value than the partitions we've removed in `dbt_klaviyo`, but I'm wondering if the config needs to be modified here to improve performance. Curious as to your thoughts here!

**Comment by fivetran-joemarkiewicz** [comment]
@fivetran-avinash good callout on the partition present in the orders attribution model. After looking into it, I see we are aggressively partitioning on the timestamp grain for a field we aren't including in our incremental logic. Therefore, removing the partition would be more appropriate in this case. See the updated validations on the latest commit.

image

image

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,20 @@
+# dbt_shopify_holistic_reporting v0.7.0
+
+[PR #26](https://github.com/fivetran/dbt_shopify_holistic_reporting/pull/26) includes the following changes:
+
+## Upstream Klaviyo and Shopify Holistic Reporting Breaking Changes (Full refresh required after upgrading)
+- Upstream and immediate incremental models within the dbt_klaviyo and dbt_shopify_holistic_reporting package running on BigQuery have had the `partition_by` logic removed. This change affects only BigQuery warehouses and resolves the `too many partitions` error that some users encountered. The partitioning was also deemed unnecessary for the aforementioned models and their downstream references, offering no performance benefit. By removing it, we eliminate both the error risk and an unneeded configuration. Refer to the [v0.8.0 dbt_klaviyo release notes](https://github.com/fivetran/dbt_klaviyo/releases/tag/v0.8.0) for more details regarding the upstream changes. This change applies to the following models:
```
Comment (fivetran-avinash) [inline CHANGELOG.md:6]:
```suggestion
- Upstream and immediate incremental models within the dbt_klaviyo and dbt_shopify_holistic_reporting packages running on BigQuery have had the `partition_by` logic removed. This change affects only BigQuery warehouses and resolves the `too many partitions` error that some users encountered. The partitioning was also deemed unnecessary for the aforementioned models and their downstream references, offering no performance benefit. By removing it, we eliminate both the error risk and an unneeded configuration. Refer to the [v0.8.0 dbt_klaviyo release notes](https://github.com/fivetran/dbt_klaviyo/releases/tag/v0.8.0) for more details regarding the upstream changes. This change applies to the following models:
```

**Review by fivetran-avinash — APPROVED** [review]
@fivetran-joemarkiewicz Thanks for the quick turnaround! One small change, but otherwise good to go. Make sure to update the packages!

**Pull request merged**